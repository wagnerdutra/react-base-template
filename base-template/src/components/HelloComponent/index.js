import React from 'react';

const HelloComponent = () => <h1>Hello World!</h1>;

export default HelloComponent;
