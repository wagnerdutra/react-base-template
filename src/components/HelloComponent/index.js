import React from 'react';
import PropTypes from 'prop-types';
import { Container } from './styles';

const HelloComponent = React.memo(({ text, userName, loading }) => (
  <Container>
    <h1>{text}</h1>
    <p>{loading ? 'Carregando...' : `Welcome: ${userName}`}</p>
  </Container>
));

HelloComponent.propTypes = {
  text: PropTypes.string.isRequired,
  userName: PropTypes.string.isRequired,
  loading: PropTypes.bool.isRequired
};

export default HelloComponent;
