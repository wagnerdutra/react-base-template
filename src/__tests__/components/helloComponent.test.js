import React from 'react';
import { shallow } from 'enzyme';

import HelloComponent from 'Components/HelloComponent';

describe('HelloComponent', () => {
  it('should render loading correctly', () => {
    const wrapper = shallow(<HelloComponent text="" userName="" loading />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component props correctly', () => {
    const wrapper = shallow(<HelloComponent text="OlÃ¡" userName="Wagner" loading={false} />);
    expect(wrapper).toMatchSnapshot();
  });
});
