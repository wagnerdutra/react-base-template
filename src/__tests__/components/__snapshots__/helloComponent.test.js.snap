// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloComponent should render component props correctly 1`] = `
<styled.div>
  <h1>
    Ol√°
  </h1>
  <p>
    Welcome: Wagner
  </p>
</styled.div>
`;

exports[`HelloComponent should render loading correctly 1`] = `
<styled.div>
  <h1 />
  <p>
    Carregando...
  </p>
</styled.div>
`;
